src/main.rs

```diff
@@ -27,72 +29,29 @@ impl Default for Snapshots {
 
 impl Snapshots {
     pub fn new() -> Self {
-        let timestamp = SystemTime::now();
+        let snapshot = metriken_exposition::Snapshotter::default().snapshot();
 
-        let mut current = HashMap::new();
-
-        for metric in metriken::metrics().iter() {
-            let any = if let Some(any) = metric.as_any() {
-                any
-            } else {
-                continue;
-            };
-
-            let key = metric.name().to_string();
-
-            let snapshot = if let Some(histogram) = any.downcast_ref::<metriken::AtomicHistogram>()
-            {
-                histogram.snapshot()
-            } else if let Some(histogram) = any.downcast_ref::<metriken::RwLockHistogram>() {
-                histogram.snapshot()
-            } else {
-                None
-            };
-
-            if let Some(snapshot) = snapshot {
-                current.insert(key, snapshot);
-            }
-        }
-
-        let deltas = current.clone();
+        let previous: HistogramSnapshots = snapshot.histograms.into_iter().collect();
+        let delta = previous.clone();
 
         Self {
-            timestamp,
-            previous: current,
-            deltas,
+            timestamp: snapshot.systemtime,
+            previous,
+            delta,
         }
     }
 
     pub fn update(&mut self) {
-        self.timestamp = SystemTime::now();
-
-        let mut current = HashMap::new();
+        let mut snapshot = metriken_exposition::Snapshotter::default().snapshot();
+        self.timestamp = snapshot.systemtime;
 
-        for metric in metriken::metrics().iter() {
-            let any = if let Some(any) = metric.as_any() {
-                any
-            } else {
-                continue;
-            };
+        let mut current: HistogramSnapshots = snapshot.histograms.drain(..).collect();
```

Comment:
I'm curious: what is the difference between `histograms.drain(..).collect()` and the `histograms.into_iter().collect()` above? Seems like they should be the same since there is no range to the drain.

Reply:
They are effectively the same and just a stylistic difference. I've corrected for consistency.